### ISSUE TITLE
Tables have some accessibility issues

### ISSUE THREAD

**Issue by greatislander** [laravel-filament/filament#4794]
### Package

filament/tables

### Package Version

v2.16.37

### Laravel Version

v9.37

### Livewire Version

v2.10.7

### PHP Version

8.1

### Problem description

I've observed two significant issues with the admin tables that make them very difficult to use for a screen reader user (and not WCAG compliant).

 1. The bulk action checkboxes do not have an accessible label.
 2. The sort buttons at the top of the column don't indicate what happens when you click them.

### Expected behavior

1. The checkbox should reference the table row it is used to select in one of three ways:

- giving each checkbox a unique `id` and providing a `<label>` which references the `id` via its `for` attribute;
- referencing another element by `id` which provides a label for the checkbox using the `aria-labelledby` attribute;
- using an `aria-label` attribute directly on the checkbox input

The checkbox should be announced as (for example): "Test User, checkbox, unchecked"

2. The sort button, in addition to the column name, should include text visually hidden using Tailwind's `.sr-only` class which says "Sort (ascending|descending) by". The button would then be announced as "Sort ascending by name" instead of just "Name", providing better context to a screen reader user.

### Steps to reproduce

Preparation: clone repository, install dependencies, run `php artisan migrate:fresh --seed`. Log in to `/admin` as user test@example.com, password = 'password'. Visit the Users page in the admin area.

1. Using VoiceOver on macOS, every checkbox on the Users admin page is announced as "unchecked checkbox" with no further context.

2. Using VoiceOver on macOS, the sort buttons on the Users admin page only give the column name and not any context for what will happen when they are pressed.

### Reproduction repository

https://github.com/greatislander/filament-tables

**Comment by greatislander** [comment]
P.S. I would be happy to contribute PRs for both of these issues.

**Comment by danharrin** [comment]
PRs would be great, thanks!

**Comment by danharrin** [comment]
Please ensure that any labels are also added as translatable

**Comment by Pen-y-Fan** [comment]
I also spotted the accessibility issues when testing to WCAG 2.1 AA standard. I will hold off raising a PR until the OP @greatislander has finished this PR. 

I can confirm the issue using the OP's **Reproduction repository**, then run Google Chrome's lighthouse report, set to test accessibility.  See [admin-users-table.pdf](https://github.com/filamentphp/filament/files/9945941/admin-users-table.pdf). The current score is 86%.

- The input is as per the OP's issue. regarding the checkbox 
- The drop-down selector for the number of pages isn't mentioned. It would be a similar fix to the input.
- Contrast ratio issues are also not raised, although they will be per the theme. As **amber** is the default, it may be worth making the default theme compliant. However, I'm not sure how this will affect other themes, do they inherit the default?

Tailwind CSS makes fixing contrast radio problems easier:

- **User** white text with amber background (bg-primary-500): increase the background by 100 until the ratio passes (e.g. bg-primary-600 > bg-primary-700). 
- **Edit** amber text (text-primary-500) on white background: increase the text by 100 until the ratio passes (e.g. text-primary-600 > text-primary-700).

I expect the CSS will need to be recompiled.

Thanks

**Comment by danharrin** [comment]
Issues addressed in #5648
